hs_core/tests/api/views/test_authorize.py

```diff
@@ -0,0 +1,408 @@
+from django.contrib.auth.models import Group, AnonymousUser
+from django.test import TestCase, RequestFactory
+from rest_framework.exceptions import PermissionDenied, NotFound
+
+from hs_core.hydroshare import resource
+from hs_core.hydroshare import users
+from hs_core.testing import MockIRODSTestCaseMixin
+from hs_core.views.utils import authorize
+from hs_access_control.models import PrivilegeCodes
+
+class TestAuthorize(MockIRODSTestCaseMixin, TestCase):
+
+    def setUp(self):
+        super(TestAuthorize, self).setUp()
+        self.group, _ = Group.objects.get_or_create(name='Hydroshare Author')
+        # create a user - resource owner
+        self.user = users.create_account(
+            'test_user@email.com',
+            username='testuser',
+            first_name='some_first_name',
+            last_name='some_last_name',
+            superuser=False,
+            groups=[])
+        self.res = resource.create_resource(
+            'GenericResource',
+            self.user,
+            'My Test Resource'
+            )
+
+        self.request = RequestFactory().request()
+
+    def test_authorize_owner(self):
+        common_parameters = [
+                      {'res_id': self.res.short_id, 'discoverable': False, 'edit': False, 'view':True, 'full': False,
```

Comment:
It would be helpful to comment each parameter permutation with a brief sentence about the intention of each assertion.  This will help us to ensure that we cover all relevant cases.